File: Java/dependency-injection/bastard-injection/src/main/java/com/github/grzesiek_galezowski/BastardInjection/Outbound/Outbound.java
package com.github.grzesiek_galezowski.BastardInjection.Outbound;

import com.github.grzesiek_galezowski.BastardInjection.Interfaces.Message;

/**
 * Created by grzes on 28.11.2016.
 */
public interface Outbound {
  void send(Message message);
}

File: CSharp/DependencyInjection/DependencyInjection/ServiceLocatorDIAntipattern/_2_ServiceLocator/Outbound/Outbound.cs
using ServiceLocatorDIAntipattern._2_ServiceLocator.Interfaces;
using Unity;

namespace ServiceLocatorDIAntipattern._2_ServiceLocator.Outbound
{
  public interface IOutbound
  {
    void Send(AcmeMessage message);
  }

  public class Outbound : IOutbound
  {
    private readonly IOutputSocket _outputOutputSocket;

    public Outbound()
    {
      _outputOutputSocket = ApplicationRoot.Context.Resolve<IOutputSocket>();
    }

    public void Send(AcmeMessage message)
    {
      var outboundMessage = ApplicationRoot.Context.Resolve<IOutboundMessage>();
      message.WriteTo(outboundMessage);
      outboundMessage.SendVia(_outputOutputSocket);
    }
  }
}